tests/cppunit/compact_test.cc

```diff
@@ -18,9 +18,9 @@
  *
  */
 
+#include <filesystem>
```

Comment:
I don't know if we should do conditional include here. But since we require C++17, it should be fine as is.